packages/utah-design-system/src/components/Sherlock.jsx

```diff
@@ -59,6 +59,30 @@ let defaultItems = [
   { id: 5, name: '84111' },
 ];
 
+async function safeFetch(url, options) {
+  let responseJson;
+  try {
+    const response = await fetch(url, options);
+
+    if (!response.ok) {
+      throw new Error(
+        `non-200 status code returned from ${url}: ${response.status} - ${response.statusText}`,
+      );
+    }
+
+    responseJson = await response.json();
+  } catch (error) {
+    throw new Error(`error thrown during fetch to ${url}: ${error.message}`);
+  }
+
+  // handle esri response errors which return a 200 status code
+  if (responseJson.error) {
+    throw new Error(`${url} returned an error: ${responseJson.error.message}`);
+  }
+
+  return responseJson;
+}
```

Comment:
Do you think there is utility in making this available to all our components via @ugrc/utilities or is it specific enough for sherlock?

Reply:
I went back and forth on this and eventually decided to keep it. But if you think that it should be moved to utilities, I'm happy to move it. I also wondered if I had just duplicated some of ky and if we should switch to that.

Reply:
Yeah. I wobble on that also. We use fetch to have less dependencies but we're duplicating the code and use ky in almost all of our projects anyway so it's likely a free dependency.

Reply:
I'll resolve this for now and let's think about migrating to ky at some point across the board.